File: dsa_leetcode_beginner/0300-longest-increasing-subsequence/0300-longest-increasing-subsequence.js

function lengthOfLIS(nums) {
    if (nums.length === 0) {
        return 0;
    }

    let dp = new Array(nums.length).fill(1);
    let max_length = 1;

    for (let i = 0; i < nums.length; i++) {
        for (let j = 0; j < i; j++) {
            if (nums[i] > nums[j]) {
                dp[i] = Math.max(dp[i], dp[j] + 1);
            }
        }
        max_length = Math.max(max_length, dp[i]);
    }

    return max_length;
}

File: dsa_leetcode_beginner/0300-longest-increasing-subsequence/0300-longest-increasing-subsequence.cpp

#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        if (nums.empty()) {
            return 0;
        }

        vector<int> dp(nums.size(), 1);
        int max_length = 1;

        for (int i = 0; i < nums.size(); ++i) {
            for (int j = 0; j < i; ++j) {
                if (nums[i] > nums[j]) {
                    dp[i] = max(dp[i], dp[j] + 1);
                }
            }
            max_length = max(max_length, dp[i]);
        }

        return max_length;
    }
};
